src/XCCDF_POLICY/xccdf_policy.c

```diff
@@ -2402,25 +2356,25 @@ struct xccdf_item * xccdf_policy_tailor_item(struct xccdf_policy * policy, struc
     xccdf_type_t type = xccdf_item_get_type(item);
     switch (type) {
         case XCCDF_RULE: {
-            struct xccdf_refine_rule * r_rule = xccdf_policy_get_refine_rules_by_rule(policy, item);
+	    struct xccdf_refine_rule_internal * r_rule = xccdf_policy_get_refine_rule_by_item(policy, item);
             if (r_rule == NULL) return item;
 
             new_item = (struct xccdf_item *) xccdf_rule_clone((struct xccdf_rule *) item);
-            if ((int)xccdf_refine_rule_get_role(r_rule) > 0)
-                xccdf_rule_set_role((struct xccdf_rule *) new_item, xccdf_refine_rule_get_role(r_rule));
-            if ((int)xccdf_refine_rule_get_severity(r_rule) > 0)
-                xccdf_rule_set_severity((struct xccdf_rule *) new_item, xccdf_refine_rule_get_severity(r_rule));
-            if (xccdf_refine_rule_weight_defined(r_rule))
-                xccdf_rule_set_weight((struct xccdf_rule *) new_item, xccdf_refine_rule_get_weight(r_rule));
-            break;
-        }
+            if (r_rule->role > 0)
+                xccdf_rule_set_role((struct xccdf_rule *) new_item, r_rule->role);
+            if (r_rule->severity > 0)
```

Comment:
`severity > XCCDF_LEVEL_NOT_DEFINED`

is more fancy. :-P (obviously just kidding)